File: Kap15/C/Listing07/ByteReader.c
#include <stdio.h>
#include <stdlib.h>

void readBytes(char *fName, int noOfBytes) {
  int readBytes = 0;
  unsigned char *myBuffer;
  FILE *fp;

  myBuffer = (unsigned char *) malloc(noOfBytes * sizeof(unsigned char));
  fp = fopen(fName,"rb");  
  readBytes = fread(myBuffer, sizeof(unsigned char), noOfBytes, fp);

  fclose(fp);
  for(int i = 0; i < readBytes; i++) {
    printf("\n%02x ", myBuffer[i]);
  }

  free(myBuffer);
}

int main() {
    readBytes("C:\\temp\\MyBitmap.bmp", 100);
    return 0;
}

File: Kap15/PHP/Listing10/ByteReader.php
<?php
function readBytes($fName, $noOfBytes) {
  $fp = fopen($fName, "rb");
  if ($fp) {
    if ($noOfBytes > filesize($fName)) {
      $noOfBytes = filesize($fName);
    }
    $myBytes = fread($fp, $noOfBytes);
    fclose($fp);
    for($i = 0; $i < strlen($myBytes); $i++) {
      echo "\n" .dechex(ord($myBytes[$i]));
    }
  }
}

readBytes("C:\\temp\\MyBitmap.bmp", 100);
?>